lib/rspec/matchers/built_in/has.rb

```diff
@@ -46,8 +46,22 @@ def description
 
       private
 
+        # Catch a semi-frequent typo - if you have strict_predicate_matchers disabled and
+        # expect(spy).to have_receieveddd(:foo) it would be evergreen - the dynamic matcher
+        # queries `has_receiveddd?`, the spy _fakes_ the method, returning its (truthy) self.
+        def really_responds_to?(method)
+          if defined?(RSpec::Mocks::Double) && @actual.is_a?(RSpec::Mocks::Double)
```

Comment:
Cosmetic. This can be optimized by extracting the defined? check outside of the method, and defining the method conditionally.
I hint at adding a ruby version check there. Don’t want to drown your time. I would wholeheartedly accept that we don’t fix that issue for users sticking to 1.8.7

Reply:
That was my suspicion, but I didn't want to waste anyone's time by checking through repeated CI runs I couldn't start. Either way, I'll need _some_ way to run the specs locally on 1.8.7 - if I don't sort it by this afternoon I'll push up a blind fix though.

Reply:
Ci would start now after you have some contributions merged.

Reply:
Ah! Okay, I'll work that way while I thrash my way through docker-based development on the side then. Ignore me until I get these suites sufficiently green -.-

Reply:
I believe that resolved it, though there's a _segfault_ on one of the legacy builds. Is that just something that happens occasionally? (I rewrote and force-pushed the last commit to prompt a rebuild; I didn't see a cleaner way to do that.)

Reply:
This happens time to time.